hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/BackupSystemTable.java

```diff
@@ -1902,4 +1905,18 @@ private static void ensureTableEnabled(Admin admin, TableName tableName) throws
       }
     }
   }
+
+  /**
+   * Executes the given operations in partitioned batches of size {@link #BATCH_SIZE}
+   */
+  private static void executePartitionedBatches(Table table, List<? extends Row> operations) {
+    List<? extends List<? extends Row>> operationBatches = Lists.partition(operations, BATCH_SIZE);
+    for (List<? extends Row> batch : operationBatches) {
+      try {
+        table.batch(batch, new Object[batch.size()]);
+      } catch (InterruptedException | IOException e) {
```

Comment:
- all caller methods already throw IOException, so it's more logical to keep IOExceptions as they are, rather than turning them into RunTimeExceptions
- InterruptedExceptions should re-propagate their interrupt.

Reply:
Agreed on the IOEx. Can you please clarify what you mean by:
> InterruptedExceptions should re-propagate their interrupt

Reply:
This is pattern I was referring to:

```
} catch (InterruptedException e) {
              Thread.currentThread().interrupt();
}
```
I see it's used in some location in HBase code, but not in others.

Basically, it comes down to how interrupts are used. If interrupts should halt the workload that is happening, it should be propagated so calling (blocking) methods of the interrupted code also get interrupted. In more specific cases, it depends...

Reply:
Fair enough 👍 made the change